force-app/main/default/classes/LeadTriggerHandlerTest.cls

```diff
@@ -0,0 +1,153 @@
+@IsTest
+
+public with sharing class LeadTriggerHandlerTest {
+
+    @IsTest
+    public static void handleTitleNormalization_Test(){
+        List<String> titleVariations = new List<String>{
+            'vp', 'v.p.', 'Vice President', 'mgr', 'manage', 'head of department', 'exec',
+            'chief', 'head', 'assist', 'deputy', 'jr'
+        };
+        
+        List<Lead> leadList = new List<Lead>();
+        
+        for (String title : titleVariations) {
+            leadList.add(new Lead(FirstName = 'Test', LastName = 'User', Title = title, Company = 'Test Company'));
+        }
+        
+        insert leadList;
```

Comment:
Although not absolutely necessary but you can add a testStart() and a testStop() around DML's